homeassistant/components/azure_data_explorer/__init__.py

```diff
@@ -0,0 +1,224 @@
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util.dt import utcnow
+
+from .client import AzureDataExplorerClient
+from .const import (
+    CONF_APP_REG_SECRET,
+    CONF_FILTER,
+    CONF_SEND_INTERVAL,
+    DATA_FILTER,
+    DATA_HUB,
+    DEFAULT_MAX_DELAY,
+    DOMAIN,
+    FILTER_STATES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
+            },
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+# fixtures for both init and config flow tests
+@dataclass
+class FilterTest:
+    """Class for capturing a filter test."""
+
+    entity_id: str
+    expect_called: bool
+
+
+async def async_setup(hass: HomeAssistant, yaml_config: ConfigType) -> bool:
+    """Activate ADX component from yaml.
+
+    Adds an empty filter to hass data.
+    Tries to get a filter from yaml, if present set to hass data.
+    If config is empty after getting the filter, return, otherwise emit
+    deprecated warning and pass the rest to the config flow.
+    """
+
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: {}})
+    if DOMAIN in yaml_config:
+        hass.data[DOMAIN][DATA_FILTER] = yaml_config[DOMAIN][CONF_FILTER]
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Do the setup based on the config entry and the filter from yaml."""
+    # hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    adx = AzureDataExplorer(hass, entry)
+    try:
+        await adx.test_connection()
+    except KustoServiceError as exp:
+        raise ConfigEntryError(
+            "Could not find Azure Data Explorer database or table"
+        ) from exp
+    except KustoAuthenticationError:
+        return False
+
+    hass.data[DOMAIN][DATA_HUB] = adx
+    entry.async_on_unload(entry.add_update_listener(async_update_listener))
+    await adx.async_start()
+    return True
+
+
+async def async_update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Update listener for options."""
+    hass.data[DOMAIN][DATA_HUB].update_options(entry.options)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    adx = hass.data[DOMAIN].pop(DATA_HUB)
+    await adx.async_stop()
+    return True
+
+
+class AzureDataExplorer:
+    """A event handler class for Azure Data Explorer."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the listener."""
+
+        self.hass = hass
+        self._entry = entry
+        self._entities_filter = hass.data[DOMAIN][DATA_FILTER]
+
+        self._client = AzureDataExplorerClient(entry.data)
+
+        self._send_interval = entry.options[CONF_SEND_INTERVAL]
+        self._client_secret = entry.data[CONF_APP_REG_SECRET]
```

Comment:
Why do we now need a secret?
Where will it be used?

Reply:
The secret is for the Azure Service Principal, it has a maximum lifetime of two years, and needs to be updated. I had planned all along to add this update, so I found it benefitial to change the the update interval to the secret.

Reply:
The secret is also a part of the initial setup

Reply:
Im looking into the issues for CI, But right now I have no idea on how to solve it. As far as I can see its not related to any code changes I made, but more to som other change. Ill be very happy for any input on how to solve the issue

Reply:
But `self._client_secret` will not used somewhere...
The community and we constantly improve HA, and the CI is failing as you need to adopt this PR to these changes. The errors are IMO clear. The return type of your functions are not matching the parent one and need to be updated

Reply:
@edenhaus  Hi, sorry to bother you. but Im having some issues that I hope you can help me with. I have solved the previous problem, but some tests are failing. its like the fixtures are not being run. I have commited the failing code. and this is the errors im seeing:

FAILED tests/components/azure_data_explorer/test_init.py::test_put_event_on_queue_with_managed_client - AttributeError: 'NoneType' object has no attribute 'args'
FAILED tests/components/azure_data_explorer/test_init.py::test_put_event_on_queue_with_managed_client_with_errors[KustoServiceError] - AssertionError: assert 'Could not find database or table' in '2024-01-01 00:00:00.000 DEBUG    MainThread homeassistant.core:core.py:1411 Bus:Handling <Event state_changed[L]: entity_id=sensor.test_sensor, old_state=None, new_state=<state sensor.test_sensor=on @ 2023-12-31T16:00:00-08:00...
FAILED tests/components/azure_data_explorer/test_init.py::test_put_event_on_queue_with_managed_client_with_errors[KustoAuthenticationError] - AssertionError: assert 'Could not authenticate to Azure Data Explorer' in '2024-01-01 00:00:00.000 DEBUG    MainThread homeassistant.core:core.py:1411 Bus:Handling <Event state_changed[L]: entity_id=sensor.test_sensor, old_state=None, new_state=<state sensor.test_sensor=on @ 2023-12-31T1...
FAILED tests/components/azure_data_explorer/test_init.py::test_put_event_on_queue_with_queueing_client - AssertionError: Expected 'ingest_from_stream' to have been called once. Called 0 times.
FAILED tests/components/azure_data_explorer/test_init.py::test_filter[allowlist] - AssertionError: assert False == True
FAILED tests/components/azure_data_explorer/test_init.py::test_filter[denylist] - AssertionError: assert False == True
FAILED tests/components/azure_data_explorer/test_init.py::test_filter[filtered_allowlist] - AssertionError: assert False == True
FAILED tests/components/azure_data_explorer/test_init.py::test_filter[filtered_denylist] - AssertionError: assert False == True

Reply:
Optionsflow is removed, so secret is gone.